pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/ViewFilter.java

```diff
@@ -688,17 +688,43 @@ public static ViewFilter parseFilteringQuery(String str, ViewFilter viewFilter)
                                 state = 9;
                                 break;
                             } else { // "LIKE %?% query
-                                value = word;
-                                if (value.contains(Filter.PERCENT_SIGN)) {
+                                if (word.contains(Filter.PERCENT_SIGN) || longWord) {
                                     exact = false;
-                                    int first = value.indexOf(Filter.PERCENT_SIGN);
-                                    int last = value.lastIndexOf(Filter.PERCENT_SIGN);
-                                    if (first != -1) {
-                                        value = value.substring(first + 1, last);
+                                    int first = word.indexOf(Filter.PERCENT_SIGN);
+                                    int last = word.lastIndexOf(Filter.PERCENT_SIGN);
+                                    if (first != -1 && first != last) {
+                                        value = word.substring(first + 1, last);
+                                        state = 11;
+                                    } else {
+                                        // "LIKE %more than 1 word% query
+                                        tempValue = word;
+                                        if (!longWord) {
+                                            if (tempValue.contains(Filter.SINGLE_QUOTE)) {
+                                                if (tempValue.indexOf(Filter.SINGLE_QUOTE) != tempValue.lastIndexOf(Filter.SINGLE_QUOTE)) {
+                                                    value = trimValue(tempValue);
+                                                    state = 11;
+                                                    break;
+                                                } else {
+                                                    longWord = true;
+                                                    value += trimValue(tempValue) + " ";
```

Comment:
i think a potential bug here would be searching for something with double spaces ( it could already be in the current code too)